### ISSUE TITLE
Best Practice: How to enhance already existing models with lightning?

### ISSUE THREAD

**Issue by maxfrei750** [PyTorchLightning/pytorch-lightning#5305]
## ❓ Questions and Help

I'd like to enhance an already existing model ([MaskRCNN](https://github.com/pytorch/vision/blob/master/torchvision/models/detection/mask_rcnn.py)) with lightning capabilities. Therefore, I'm wondering what the best practice might be.

From what I've seen, there are three possibilities:

1. Copy all necessary code and use `pl.LightningModule` instead of `nn.Module` as base class. 
2. Create a new class and have the MaskRCNN model as an attribute (e.g. `self.model`).
3. Use multi-inheritance to combine `pl.LightningModule` and `MaskRCNN`: 
    ```
    class LightningMaskRCNN(MaskRCNN, pl.LightningModule)
        ...
    ```
IMHO, 1 is the solution closest to the [documentation](https://pytorch-lightning.readthedocs.io/en/latest/lightning_module.html#lightning-module), but obviously no good solution for the given use-case, since it uses duplicated code. 2 and 3 are both possible (if I didn't overlook something) but I can't decide which one to use (although I tend to 3), so I was wondering if there is a best practice for these use-cases.

Thanks!

**Comment by s-rog** [comment]
It all comes down to what's best for your workflow as lightning allows for a lot of flexibility. I started using lightning with 1, but now I find myself using 2 more.

**Comment by DuinoDu** [comment]
The user case is to finetune exsiting pytorch model, which may not be trained in lightning, right？
In my opinion, LightningModule wraps network forward, loss, optimimizer, learning rate. 
`3` might be consise, but may include multi-inherite problem.
`2` is simple, but might include some extra code.
`1` is not suggested, except you need to change much inner maskrcnn behaviors.

Peresonally prefer to `2`.

**Comment by maxfrei750** [comment]
> The user case is to finetune exsiting pytorch model, which may not be trained in lightning, right？

Sort of. The main goal is to reuse official [torchvision](https://github.com/pytorch/vision) code to train models using lightning.

Thanks for your inputs. Since there does not seem to be an official best practice, I think I will just try to get 3 to work and resort to 2 if I run into problems.

**Comment by tchaton** [comment]
Dear @maxfrei750,

I think it is better to use 2, as you can just create a `LitTorchVisionModel`.

Best regards,
T.C

**Comment by tchaton** [comment]
Closing this issue for now.
